### DISCUSSION TITLE
Removing v0.35.x releases

### DISCUSSION THREAD

**Discussion by thanethomson** [tendermint/tendermint#9155] [Announcements]
As per [this blog post](https://medium.com/the-interchain-foundation/discontinuing-tendermint-v0-35-a-postmortem-on-the-new-networking-layer-3696c811dabc), all of the v0.35.x releases have been removed.

We will be skipping the v0.36.x series and going straight to v0.37, which branches off from the v0.34 line.

**Comment by faustbrian** [comment]
Any ETA on when a v0.37 release will be available? Our application was downloading https://github.com/tendermint/tendermint/releases/download/v0.35.9/tendermint_0.35.9_linux_amd64.tar.gz and that started to fail on deployments this morning and downgrading to 0.34 would be quite painful because of the differences in protobuf files.

**Comment by faustbrian** [comment]
Looks like Homebrew can still pull in those deleted releases because the tags still exist for all 0.35 releases, for example https://github.com/tendermint/tendermint/releases/tag/v0.35.9.

```
❯ brew install tendermint
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
Updated 2 taps (homebrew/core and homebrew/cask).
==> New Formulae
berkeley-db@5         cpp-httplib           gaze                  onlykey-agent         prr                   sgr                   smap
==> New Casks
duckduckgo                                                                   v2ray-unofficial

You have 4 outdated formulae installed.
You can upgrade them with brew upgrade
or list them with brew outdated.

==> Downloading https://ghcr.io/v2/homebrew/core/tendermint/manifests/0.35.9
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/tendermint/blobs/sha256:2d0f6f119ae6ad63d371f13d02bb6a8e4be6c4024ba5a191a6529fed4160d780
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:2d0f6f119ae6ad63d371f13d02bb6a8e4be6c4024ba5a191a6529fed4160d780?se=
######################################################################## 100.0%
==> Pouring tendermint--0.35.9.arm64_monterey.bottle.tar.gz
🍺  /opt/homebrew/Cellar/tendermint/0.35.9: 6 files, 25.2MB
==> Running `brew cleanup tendermint`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```

**Comment by thanethomson** [comment]
We're hoping to cut the v0.37 release toward the end of Q3 of 2022.

At present we're not planning an upgrade path from v0.35 to v0.37, and given that v0.37 will be based on v0.34, it's probably a good idea to look into migrating to v0.34. The only upgrade path we're currently looking at supporting for v0.37 is from v0.34.

We also plan on ensuring that we don't break the protos across releases in future. Protobuf changes from v0.37 onwards will be additive relative to v0.34.

Please let us know if there's anything specific you'd need assistance with in migrating?

**Comment by faustbrian** [comment]
Alright, will look into downgrading to 0.34 then. The biggest change we noticed with proto files from 0.34 compared to 0.35 is that the events expect the `key` and `value` properties to be buffers instead of strings. We use TypeScript so the proto files result in files that we use as type hints and downgrading to 0.34 causes 800+ errors in our application because we need to change all `key` and `value` properties to be buffers instead of strings and some other bits but they seemed less invasive than those.

<img width="389" alt="CleanShot 2022-08-04 at 05 53 49@2x" src="https://user-images.githubusercontent.com/22145591/182752904-95a6017e-3f93-4a48-a944-a8549ea1b2cf.png">

Was there any reason for changing those properties to strings in 0.35 and is this something that is going to be ported to 0.37? The above screenshot is the diff from downgrading to 0.34 from the 0.35.9 release.

**Reply by Jiuhk** [reply]
Please take a look at this issue: https://github.com/tendermint/tendermint/issues/6403.

I also want to ask is this something that is going to be ported to 0.37, but I'm looking forward to that it does, because it fixes a major problem I'm having in my application.

**Comment by faustbrian** [comment]
The `mode` parameter doesn't seem to exist in the `config.toml` of the 0.34.x releases. What are the equivalent configuration values in 0.34.x to achieve the same behaviours as with `full`, `validator` and `seed` in 0.35.x?

```toml
# Mode of Node: full | validator | seed (default: "validator")
# * validator node (default)
#   - all reactors
#   - with priv_validator_key.json, priv_validator_state.json
# * full node
#   - all reactors
#   - No priv_validator_key.json, priv_validator_state.json
# * seed node
#   - only P2P, PEX Reactor
#   - No priv_validator_key.json, priv_validator_state.json
mode = "validator"
```

**Edit**
Looks like the comments reflect which files have to be present or absent in 0.34 while in 0.35 it is dictated by the `mode` flag.

**Reply by cmwaters** [reply]
In v0.34 there is no distinction between `full` and `validator`. In v0.34 you need to set `SeedMode` to `true` to turn a full node into a seed node
